modules/apps/commerce/commerce-checkout-web/src/main/resources/META-INF/resources/checkout_step/address.jsp

```diff
@@ -64,16 +64,36 @@ boolean hasManageAddressesPermission = baseAddressCheckoutStepDisplayContext.has
 
 					<%
 					boolean addressWasFound = false;
+					boolean countryChannelRelExists = false;
 
 					for (CommerceAddress commerceAddress : commerceAddresses) {
+						Country country = commerceAddress.getCountry();
+
 						boolean selectedAddress = commerceAddressId == commerceAddress.getCommerceAddressId();
 
+						List<CommerceChannelRel> commerceChannelRels = CommerceChannelRelLocalServiceUtil.getCommerceChannelRels(Country.class.getName(), commerceAddress.getCountryId(), null, -1, -1);
+
+						for (CommerceChannelRel commerceChannelRel : commerceChannelRels) {
+							if (commerceChannelRel.getCommerceChannelId() == commerceContext.getCommerceChannelId()) {
+								countryChannelRelExists = true;
+							}
+						}
+
```

Comment:
move this logic to getAddresses method in the Context class so that the filtered list is returned

Reply:
```
		if (commerceAddresses != null) {
			List<CommerceAddress> filteredCommerceAddresses = new ArrayList<>();

			for (CommerceAddress commerceAddress : commerceAddresses) {
				Country country = commerceAddress.getCountry();

				if (country.isGroupFilterEnabled()) {
					List<CommerceChannelRel> commerceChannelRels =
						commerceChannelRelLocalService.getCommerceChannelRels(
							Country.class.getName(),
							commerceAddress.getCountryId(), null, -1, -1);

					for (CommerceChannelRel commerceChannelRel :
							commerceChannelRels) {

						if (commerceChannelRel.getCommerceChannelId() ==
								_commerceContext.getCommerceChannelId()) {

							filteredCommerceAddresses.add(commerceAddress);
						}
					}
				}
				else {
					filteredCommerceAddresses.add(commerceAddress);
				}
			}

			return filteredCommerceAddresses;
		}
```

Reply:
@andrea-ale-sbarra , would you be okay with this? We would be filtering after we get the results from:
```
		List<CommerceAddress> commerceAddresses =
			commerceAddressService.getCommerceAddressesByCompanyId(
				_commerceOrder.getCompanyId(), AccountEntry.class.getName(),
				_commerceOrder.getCommerceAccountId(), QueryUtil.ALL_POS,
				QueryUtil.ALL_POS, new CommerceAddressNameComparator());
```